File: frontend/src/api/skill.ts
import request from '../utils/request'

export interface Skill {
  id: number
  skillName: string
  skillCode: string
  categoryId: number
  categoryName: string
  description: string
  inputExample: string
  outputExample: string
  author: string
  version: string
  status: number
  viewCount: number
  useCount: number
  createTime: string
  tags: string[]
  // 新增字段
  type: number
  deptId: number
  deptName: string
  sourceType: number
  sourceDesc: string
  sourceLink: string
  tutorialUrl: string
  filePath: string
  screenshots: string[]
  visibility: number
  needAuth: number
  authApproverId: number
  isFeatured: number
  downloadCount: number
  avgRating: number
  rejectReason: string
  projectIds: number[]
}

export interface SkillDTO {
  id?: number
  skillName: string
  skillCode: string
  categoryId: number
  description?: string
  inputExample?: string
  outputExample?: string
  author?: string
  version?: string
  status?: number
  // 新增字段
  type?: number
  deptId?: number
  sourceType?: number
  sourceDesc?: string
  sourceLink?: string
  tutorialUrl?: string
  filePath?: string
  screenshots?: string[]
  visibility?: number
  needAuth?: number
  authApproverId?: number
  projectIds?: number[]
}

export function getSkillList(params: { page: number; pageSize: number; keyword?: string; categoryId?: number; status?: number; type?: number }) {
  return request.get<any, { data: { records: Skill[]; total: number } }>('/skill/list', { params })
}

export function getSkillById(id: number) {
  return request.get<any, { data: Skill }>(`/skill/${id}`)
}

export function createSkill(data: SkillDTO) {
  return request.post<any, { data: number }>('/skill', data)
}

export function updateSkill(data: SkillDTO) {
  return request.put('/skill', data)
}

export function deleteSkill(id: number) {
  return request.delete(`/skill/${id}`)
}

export function updateSkillTags(id: number, tags: string[]) {
  return request.post(`/skill/${id}/tags`, tags)
}

export function approveSkill(id: number) {
  return request.post(`/skill/${id}/approve`)
}

export function rejectSkill(id: number, reason: string) {
  return request.post(`/skill/${id}/reject`, { reason })
}

export function offlineSkill(id: number, reason: string) {
  return request.post(`/skill/${id}/offline`, { reason })
}

export function downloadSkill(id: number) {
  return request.post(`/skill/${id}/download`)
}

File: backend/src/main/java/com/yunbei/skill/entity/Skill.java
package com.yunbei.skill.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("skill")
public class Skill {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String skillName;

    private String skillCode;

    private Long categoryId;

    private String description;

    private String inputExample;

    private String outputExample;

    private String author;

    private String version;

    private Integer status;

    private Integer viewCount;

    private Integer useCount;

    private Long createBy;

    // 新增字段
    private Integer type;

    private Long deptId;

    private Integer sourceType;

    private String sourceDesc;

    private String sourceLink;

    private String tutorialUrl;

    private String filePath;

    private String screenshots;

    private Integer visibility;

    private Integer needAuth;

    private Long authApproverId;

    private Integer isFeatured;

    private Integer downloadCount;

    private BigDecimal avgRating;

    private String rejectReason;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}